packages/legend-query-builder/src/components/QueryBuilder.tsx

```diff
@@ -405,244 +281,46 @@ export const QueryBuilder = observer(
             isLoading={queryBuilderState.resultState.exportState.isInProgress}
           />
           <div className="query-builder__content">
-            <div className="query-builder__header">
-              <div className="query-builder__header__statuses">
-                {queryBuilderState.watermarkState.value && (
-                  <button
-                    className="query-builder__header__status query-builder__header__status--action"
-                    onClick={openWatermark}
-                    tabIndex={-1}
-                    title="Used watermark"
-                    name="Used watermark"
-                  >
-                    <WaterDropIcon />
-                  </button>
-                )}
-                {queryBuilderState.isCalendarEnabled && (
-                  <div
-                    className="query-builder__header__status"
-                    title="Used calendar aggregation"
-                  >
-                    <CalendarClockIcon className="query-builder__header__status__icon--calendar" />
-                  </div>
-                )}
-                {queryBuilderState.watermarkState.isEditingWatermark && (
-                  <QueryBuilderWatermarkEditor
-                    queryBuilderState={queryBuilderState}
-                  />
-                )}
-              </div>
-              <div className="query-builder__header__actions">
-                <div className="query-builder__header__actions__undo-redo">
-                  <UndoButton
-                    parent={queryBuilderRef}
-                    canUndo={
-                      queryBuilderState.changeHistoryState.canUndo &&
-                      queryBuilderState.isQuerySupported
-                    }
-                    undo={undo}
-                  />
-                  <RedoButton
-                    parent={queryBuilderRef}
-                    canRedo={
-                      queryBuilderState.changeHistoryState.canRedo &&
-                      queryBuilderState.isQuerySupported
-                    }
-                    redo={redo}
-                  />
+            {!hideQueryBuilderHeader && (
```

Comment:
seems like we are over doing it here with extensions. can we please keep the core actions directly on this editor.
Then lets think about how to separate the different buttons coming in. 

Maybe we can group the saved query ones into an extension which would include, productize query, dev service, rename query, new query, and a loader ?
that way if something wants to provide this stored query extension, they would  need to provide those items.